src/services/tag.service.ts

```diff
@@ -1,29 +1,55 @@
 import { Service } from "../services/service";
 import { Tag } from "../model/tag";
-import { Section } from "../model/section";
 import { Injectable } from "../utility/service-gen";
 
 @Injectable
 export class TagService extends Service {
 
-    tags: Tag[] = [];
-    section: Section;
+    private _tags: Tag[];
+    private updateAgents: Function[];
     
     constructor(t: string) {
         super(t);
-        this.tags = [];
+        this._tags = [];
+        this.updateAgents = [];
     }
 
     addTag(tag: Tag) {
-        this.tags.push(tag);
+        this._tags.push(tag);
+        this.update();
     }
 
     removeTag(tag: Tag) {
-        const index = this.tags.indexOf(tag);
-        if (index > -1)
-            this.tags.splice(index, 1);
+        const index = this._tags.indexOf(tag);
+        if (index > -1){
+            this._tags.splice(index, 1);
+            this.update();
+        }
         else
             throw ("Tag Not Found.");
     }
 
-}
\ No newline at end of file
+    get tags(): Tag[]{
+        return this._tags;
+    }
+
+    public valueSelect(name: string): Tag{
```

Comment:
What does this method do? Please either add comments or use a more descriptive name.